test/Donations.spec.ts

```diff
@@ -0,0 +1,379 @@
+import type { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { ethers } from "hardhat";
+import { expect } from "chai";
+import { time } from "@openzeppelin/test-helpers";
+import { BigNumber, utils } from "ethers";
+
+import type { Donations } from "../typechain";
+import { MockDAI, MockDAI__factory } from "../typechain";
+import { donationParams } from "./shared/factories";
+import { getLastBlockTimestamp, moveForwardTwoWeeks } from "./shared";
+
+const { parseUnits } = ethers.utils;
+
+const DUMMY_TX =
+  "0x5bdfd14fcc917abc2f02a30721d152a6f147f09e8cbaad4e0d5405d646c5c3e1";
+
+const CHARITY_ID = 1;
+
+describe("Donations", () => {
+  let owner: SignerWithAddress;
+  let alice: SignerWithAddress;
+  let bob: SignerWithAddress;
+  let carol: SignerWithAddress;
+  let donations: Donations;
+  let underlying: MockDAI;
+  let WORKER_ROLE: string;
+  let DEFAULT_ADMIN_ROLE: string;
+
+  beforeEach(async () => {
+    [owner, alice, bob, carol] = await ethers.getSigners();
+
+    const Donations = await ethers.getContractFactory("Donations");
+    const MockDAI = await ethers.getContractFactory("MockDAI");
+
+    underlying = (await MockDAI.deploy(0)) as MockDAI;
+    donations = (await Donations.deploy(owner.address)) as Donations;
+    WORKER_ROLE = await donations.WORKER_ROLE();
+    DEFAULT_ADMIN_ROLE = await donations.DEFAULT_ADMIN_ROLE();
+  });
+
+  describe("setTTL", () => {
+    it("changes the TTL", async () => {
+      const newTTL = BigNumber.from(time.duration.days(100).toNumber());
+
+      expect(await donations.ttl()).not.to.equal(newTTL);
+
+      await donations.setTTL(newTTL);
+
+      expect(await donations.ttl()).to.equal(newTTL);
+    });
+
+    it("emits an event", async () => {
+      const newTTL = BigNumber.from(time.duration.days(100).toNumber());
+
+      const tx = donations.setTTL(newTTL);
+
+      await expect(tx).to.emit(donations, "TTLUpdated").withArgs(newTTL);
+    });
+
+    it("fails if the caller is not authorized", async () => {
+      const newTTL = BigNumber.from(time.duration.days(100).toNumber());
+
+      await expect(donations.connect(alice).setTTL(newTTL)).to.be.revertedWith(
+        `AccessControl: account ${alice.address.toLocaleLowerCase()} is missing role ${DEFAULT_ADMIN_ROLE}`
+      );
+    });
+  });
+
+  describe("donate", () => {
+    it("transfers the available amount in the given token to the charity", async () => {
+      // set donated amount
+      await underlying.mint(donations.address, parseUnits("200"));
+
+      // create donations
+      await donations.mint(DUMMY_TX, 0, [
+        donationParams.build({
+          destinationId: CHARITY_ID,
+          amount: parseUnits("100"),
+          owner: alice.address,
+          token: underlying.address,
+        }),
+        donationParams.build({
```

Comment:
this case does not make sense to me, because the donationParams in the array are same.
Can we use different params? ex, different owner, different amount, etc.